lib/brcobranca/calculo.rb

```diff
@@ -41,6 +41,48 @@ def modulo11_2to9
       [0, 10, 11].include?(valor) ? 1 : valor
     end
 
+    # Calcula módulo 11 com multiplicadores de 2 a 9 e 2 a 5 (Utilizado pelo Santander).
+    #
+    # @return [Integer]
+    def modulo11_santander
+      return 0 if self.to_i == 0
+
+      somatorio = 0
+      multiplicadores = [2, 3, 4, 5, 6, 7, 8, 9, 2, 3, 4, 5]
+      base = "#{self.rjust(12, '0')}".reverse
+      index = 0
+
+      base.each_char do |char|
+        somatorio += base[index].to_i * multiplicadores[index].to_i
+        index += 1
+      end
+
+      resto = somatorio % 11
+      return 1 if resto == 10
+      return 0 if resto == 1 || resto == 0
+      return 11 - resto
+    end
+
+    # Calcula módulo 11 com multiplicadores de 2, 7 a 2, e 7 a 2 (Utilizado pelo Bradesco).
+    #
+    # @return [Integer, String] Caso o resto seja 1, retorna P
+
+    def modulo11_bradesco
+      somatorio = 0
+      multiplicadores = [2, 7, 6, 5, 4, 3, 2, 7, 6, 5, 4, 3, 2]
+      index = 0
+
+      self.each_char do |char|
```

Comment:
Unused block argument - `char`. You can omit the argument if you don't care about it.<br>Redundant `self` detected.